app/javascript/components/student/Editor.tsx

```diff
@@ -143,6 +150,13 @@ export function Editor({
   instructions: string
   exampleSolution: string
 }) {
+  const [theme, setTheme] = useState('vs')
+  const [keybindings, setKeybindings] = useState<Keybindings>(
+    Keybindings.DEFAULT
+  )
+  const [wrap, setWrap] = useState<'off' | 'on' | 'wordWrapColumn' | 'bounded'>(
```

Comment:
Is there a way to write this better? The `off' | 'on' | 'wordWrapColumn' | 'bounded'` is expected by Monaco.

Reply:
@SleeplessByte?

Reply:
Extract a type

Reply:
In particular: in `Settings.tsx`, I would start by writing:

```typescript
export type SettingsProps = {
  ...
}

export function Settings({ ... }: SettingsProps) {
 ...
}
```

You could then use:

```typescript
... useState<SettingsProps['wrap']>('...')
```

Need it in more places?

```typescript
export type WrapSetting = 'off' | 'on' | 'wordWrapColumn' | 'bounded'
export type SettingsProps = {
  ...
  wrap: WrapSetting 
}

...

... useState<WrapSetting>('...')
```